Shopify/cli
# Install bundler gems in local folder

### WHY are these changes introduced?

When we use the embedded CLI2, bundler installs gems in system wide folders, such as:
```
/opt/homebrew/lib/ruby/gems/3.2.0
```

This can cause issues when users upgrade their system gems, or mess with them.

### WHAT is this pull request doing?

Install all gems needed by CLI2 in it's own folder:
```
/opt/homebrew/Cellar/shopify-cli/3.44.1/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/ruby
```

### How to test your changes?

To simulate the behaviour of this patch:
- cd into `/opt/homebrew/Cellar/shopify-cli/3.44.1/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby`
- there should be no `ruby` folder
- run `/opt/homebrew/opt/ruby/bin/bundle config set --local path .`
- now from another terminal, go to a theme folder run `shopify theme check --verbose`
- after it completes successfully, the `ruby` folder should appear in `/opt/homebrew/Cellar/shopify-cli/3.44.1/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby`

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
